include/pmacc/memory/buffers/DeviceBuffer.hpp

```diff
@@ -36,365 +36,279 @@ namespace pmacc
+    /** N-dimensional device buffer
      *
-     * @tparam TYPE datatype of the buffer
-     * @tparam DIM dimension of the buffer
+     * @tparam T_Type datatype of the buffer
+     * @tparam T_dim dimension of the buffer
      */
-    template<class TYPE, unsigned DIM>
-    class DeviceBuffer : public Buffer<TYPE, DIM>
+    template<class T_Type, unsigned T_dim>
+    class DeviceBuffer : public Buffer<T_Type, T_dim>
     {
+        using BufferType = ::alpaka::Buf<ComputeDevice, T_Type, AlpakaDim<DIM1>, MemIdxType>;
+        using ViewType = alpaka::ViewPlainPtr<ComputeDevice, T_Type, AlpakaDim<T_dim>, MemIdxType>;
+        using CurrentSizeBufferDevice = ::alpaka::Buf<ComputeDevice, size_t, AlpakaDim<DIM1>, MemIdxType>;
+
+        void createSizeOnDeviceBuffers()
+        {
+            currentSizeBufferDevice.emplace(alpaka::allocBuf<size_t, MemIdxType>(
+                manager::Device<ComputeDevice>::get().current(),
+                MemSpace<DIM1>(1).toAlpakaMemVec()));
+        }
+
     public:
-        using DataBoxType = typename Buffer<TYPE, DIM>::DataBoxType;
+        using DataBoxType = typename Buffer<T_Type, T_dim>::DataBoxType;
+        std::optional<BufferType> devBuffer;
+        std::optional<ViewType> view;
+        std::optional<CurrentSizeBufferDevice> currentSizeBufferDevice;
 
-        /** Create device buffer
-         *
-         * Allocate new memory on the device.
+        using BufferType1D = ::alpaka::ViewPlainPtr<ComputeDevice, T_Type, AlpakaDim<DIM1>, MemIdxType>;
+
+        BufferType1D as1DBuffer()
+        {
+            auto currentSize = this->getCurrentSize();
+            eventSystem::startOperation(ITask::TASK_DEVICE);
+            return BufferType1D(
+                alpaka::getPtrNative(*view),
+                alpaka::getDev(*devBuffer),
+                MemSpace<DIM1>(currentSize).toAlpakaMemVec());
+        }
+
+        BufferType1D as1DBufferNElem(size_t const numElements)
+        {
+            PMACC_ASSERT(numElements < this->getCurrentSize());
+            eventSystem::startOperation(ITask::TASK_DEVICE);
+            return BufferType1D(
+                alpaka::getPtrNative(*view),
+                alpaka::getDev(*devBuffer),
+                MemSpace<DIM1>(numElements).toAlpakaMemVec());
+        }
+
+        ViewType getAlpakaView() const
+        {
+            eventSystem::startOperation(ITask::TASK_DEVICE);
+            return *view;
+        }
+
+        /** allocate data accessible from the device
          *
          * @param size extent for each dimension (in elements)
          * @param sizeOnDevice memory with the current size of the grid is stored on device
-         * @param useVectorAsBase use a vector as base of the array (is not lined pitched)
-         *                      if true size on device is atomaticly set to false
+         *
+         * @attention offset + size must be leather equal to the size of the source buffer
```

Comment:
```suggestion
         * @attention offset + size must be greater or equal to the size of the source buffer
```
Is it that what you meant?